Backend (metrics!) linked + view result refactor

mendoxei opened the pull request:
**lib version: fairreckitlib==0.4.1**

BACKEND LINK:
- supports newest version 0.4.1
- this means evaluations are linked
- arbitrary dataset matrix is used
- rating conversion linked

UPDATE 1
- some new experiment layout fixes

UPDATE 2
- VIEW RESULT FROM QUEUE, refactored result by id request to separate script

UPDATE 3
- COPY RESULT SETTINGS to new experiment (it's still a bit ugly, and **doesn't work** when doing an exp with the **mock buttons**)

mendoxei committed:
use correct conversion format for config

mendoxei committed:
Merge branch 'dev' into experiment-tab

mendoxei committed:
Merge branch 'dev' into experiment-tab

mendoxei committed:
add dataset samples, support new backend

mendoxei committed:
Merge branch 'dev' into experiment-tab

mendoxei committed:
evaluations get set when exp finishes, cleanup

mendoxei committed:
updated samples (new: LFM-1B)

mendoxei committed:
using bootstrap icon carets

 instead of fugly arrows

mendoxei committed:
use enter for tags

prevent enter for form submit

DiedevanderHoorn reviewed (changes requested):
![image](https://user-images.githubusercontent.com/35234641/170702109-326b42eb-3093-4ff2-94f4-c20ed7cadfab.png)

Only showing metrics for the second table for some reason?

mendoxei committed:
Merge branch 'dev' into experiment-tab

mendoxei commented:
@DiedevanderHoorn 
![image](https://user-images.githubusercontent.com/14370250/170728915-fad5d6b3-bf2a-45a4-af80-d11874bbc873.png)
I've tried 5 different experiments, always get metrics. Not sure what's up.

mendoxei committed:
view result from queue, refactor

refactored result getting to separate script

mendoxei committed:
CLIENT: toast fix, copy settings, dict display

copying settings doesn't work for mock buttons!!! (it's a bit hacky)
the dict display is kind of shit, i was mostly havin some good ol fun

mendoxei committed:
Merge branch 'dev' into experiment-tab

mendoxei committed:
merge (column width, button conditions)

DiedevanderHoorn reviewed (approved):
Everything seems to work now! And yay for adding view result button :) One minor thing that I thought of, maybe would be nice to only enable the view result button once a result is done (right now you can always click on it, it just does nothing sometimes so it's not an issue but might be better for the usability to grey it out when there is no result yet)
